ecs-agent/netlib/platform/managed_linux.go

```diff
@@ -0,0 +1,138 @@
+func (m *managedLinux) BuildTaskNetworkConfiguration(
+	taskID string,
+	taskPayload *ecsacs.Task,
+) (*tasknetworkconfig.TaskNetworkConfig, error) {
+	mode := aws.StringValue(taskPayload.NetworkMode)
+	var netNSs []*tasknetworkconfig.NetworkNamespace
+	var err error
+	switch mode {
+	case ecs.NetworkModeAwsvpc:
+		netNSs, err = m.common.buildAWSVPCNetworkNamespaces(taskID, taskPayload, false, nil)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to translate network configuration")
+		}
+	case ecs.NetworkModeHost:
+		netNSs, err = m.buildDefaultNetworkNamespace(taskID)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create network namespace with host eni")
+		}
+	default:
+		return nil, errors.New("invalid network mode: " + mode)
+	}
+	return &tasknetworkconfig.TaskNetworkConfig{
+		NetworkNamespaces: netNSs,
+		NetworkMode:       mode,
+	}, nil
+}
+
+func (m *managedLinux) CreateDNSConfig(taskID string,
+	netNS *tasknetworkconfig.NetworkNamespace) error {
+	return m.common.createDNSConfig(taskID, false, netNS)
+}
+
+func (m *managedLinux) ConfigureInterface(
+	ctx context.Context,
+	netNSPath string,
+	iface *networkinterface.NetworkInterface,
+	netDAO netlibdata.NetworkDataClient,
+) error {
+	return m.common.configureInterface(ctx, netNSPath, iface, netDAO)
+}
+
+func (m *managedLinux) ConfigureAppMesh(ctx context.Context,
+	netNSPath string,
+	cfg *appmesh.AppMesh) error {
+	return m.common.configureAppMesh(ctx, netNSPath, cfg)
+}
+
+func (m *managedLinux) ConfigureServiceConnect(
+	ctx context.Context,
+	netNSPath string,
+	primaryIf *networkinterface.NetworkInterface,
+	scConfig *serviceconnect.ServiceConnectConfig,
+) error {
+	return m.common.configureServiceConnect(ctx, netNSPath, primaryIf, scConfig)
+}
+
+// buildDefaultNetworkNamespace generate a default network namespace for host mode where we do not need to create new namespaces
+func (m *managedLinux) buildDefaultNetworkNamespace(taskID string) ([]*tasknetworkconfig.NetworkNamespace, error) {
+	privateIpv4, err1 := m.client.GetMetadata(PrivateIPv4Resource)
+	macAddress, err2 := m.client.GetMetadata(MacResource)
+	ec2ID, err3 := m.client.GetMetadata(InstanceIDResource)
+	subNet, err4 := m.client.GetMetadata(fmt.Sprintf(SubNetCidrBlock, macAddress))
+	macToNames, err5 := m.common.interfacesMACToName()
+	if err := goErr.Join(err1, err2, err3, err4, err5); err != nil {
+		return nil, err
+	}
+
+	hostENI := &ecsacs.ElasticNetworkInterface{
+		AttachmentArn: aws.String("arn"),
+		Ec2Id:         aws.String(ec2ID),
+		Ipv4Addresses: []*ecsacs.IPv4AddressAssignment{
+			{
+				Primary:        aws.Bool(true),
+				PrivateAddress: aws.String(privateIpv4),
+			},
+		},
+		SubnetGatewayIpv4Address:     aws.String(subNet),
+		MacAddress:                   aws.String(macAddress),
+		DomainNameServers:            []*string{},
+		DomainName:                   []*string{},
+		PrivateDnsName:               aws.String("default"),
```

Comment:
nit: should abstract this string `default` to a const

Reply:
created const for defaultArg